### ISSUE TITLE
weird ocamlyacc comments output in 4.10.2 (Apple M1)

### ISSUE THREAD

**Issue by bobzhang** [ocaml/ocaml#10490]
mly file
```
%{
    open F 
    (* fck  
        hello  why *)
%}


%token EOF
%type <unit> main
%start main
%%
main:  EOF {print_endline "hello"}
;

%%
```

generated file has weird comments: -- (fck hello why turned into f h w)
```
type token =
  | EOF

open Parsing;;
let _ = parse_error;;
# 2 "test.mly"
    open F 
    (* f  
        h  w *)
# 11 "test.ml"
let yytransl_const = [|
    0 (* EOF *);
    0|]

let yytransl_block = [|
    0|]

let yylhs = "\255\255\
\001\000\000\000"

let yylen = "\002\000\
\001\000\002\000"

let yydefred = "\000\000\
\000\000\000\000\001\000\002\000"

let yydgoto = "\002\000\
\004\000"

let yysindex = "\255\255\
\001\000\000\000\000\000\000\000"

let yyrindex = "\000\000\
\000\000\000\000\000\000\000\000"

let yygindex = "\000\000\
\000\000"

let yytablesize = 1
let yytable = "\001\000\
\003\000"

let yycheck = "\001\000\
\000\000"

let yynames_const = "\
  EOF\000\
  "

let yynames_block = "\
  "

let yyact = [|
  (fun _ -> failwith "parser")
; (fun __caml_parser_env ->
    Obj.repr(
# 12 "test.mly"
           (print_endline "hello")
# 60 "test.ml"
               : unit))
(* Entry main *)
; (fun __caml_parser_env -> raise (Parsing.YYexit (Parsing.peek_val __caml_parser_env 0)))
|]
let yytables =
  { Parsing.actions=yyact;
    Parsing.transl_const=yytransl_const;
    Parsing.transl_block=yytransl_block;
    Parsing.lhs=yylhs;
    Parsing.len=yylen;
    Parsing.defred=yydefred;
    Parsing.dgoto=yydgoto;
    Parsing.sindex=yysindex;
    Parsing.rindex=yyrindex;
    Parsing.gindex=yygindex;
    Parsing.tablesize=yytablesize;
    Parsing.table=yytable;
    Parsing.check=yycheck;
    Parsing.error_function=parse_error;
    Parsing.names_const=yynames_const;
    Parsing.names_block=yynames_block }
let main (lexfun : Lexing.lexbuf -> token) (lexbuf : Lexing.lexbuf) =
   (Parsing.yyparse yytables 1 lexfun lexbuf : unit)
;;

```

**Comment by dra27** [comment]
This was caused by #1932 in 4.10.0 and fixed in 4.11.0 by #9166.

**Comment by bobzhang** [comment]
Thanks for the pointer. would it be possible to cherry pick that for 4.10.*?
Note only 2 versions provide M1 backend, 4.10 and 4.12. 4.12 is a bit too bleeding edge, it seems also have a regression for Marshal performance(I welcome this change for good reasons but would like to pay it when multicore landed).

**Comment by dra27** [comment]
We don’t normally back-port further than the previous release (other than to fix problems caused by newer tool chains) - The release of 4.10.2 during the 4.12 dev cycle was exceptional.

I’m afraid that to use a relatively bleeding-edge platform, you may have to use a relatively bleeding-edge OCaml!

**Comment by bobzhang** [comment]
understood. I can live with that using some hacks. when I noticed this yesterday, I feel my computer may be hacked, how could this happen..?
but for other people who relies on ocamlyacc, 4.10 seems a major regression.

**Comment by gasche** [comment]
If you maintain your own variant of 4.10 for Rescript purposes, could you backport the fix from 4.11? It shouldn't be very hard as there is not a lot of change in the ocamlyacc codebase.

**Comment by bobzhang** [comment]
@gasche I can spend some time on it, what would it be called? 4.10.3?
